src/PrestaShopBundle/Form/Admin/Configure/AdvancedParameters/Employee/EmployeeType.php

```diff
@@ -0,0 +1,265 @@
+use PrestaShopBundle\Translation\TranslatorAwareTrait;
+use Symfony\Component\Form\AbstractType;
+use Symfony\Component\Form\Extension\Core\Type\ChoiceType;
+use Symfony\Component\Form\Extension\Core\Type\EmailType;
+use Symfony\Component\Form\Extension\Core\Type\PasswordType;
+use Symfony\Component\Form\Extension\Core\Type\TextType;
+use Symfony\Component\Form\FormBuilderInterface;
+use Symfony\Component\Form\FormInterface;
+use Symfony\Component\Form\FormView;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+use Symfony\Component\Validator\Constraints\Email;
+use PrestaShop\PrestaShop\Core\Domain\Profile\Employee\ValueObject\Email as EmployeeEmail;
+use Symfony\Component\Validator\Constraints\Length;
+use Symfony\Component\Validator\Constraints\NotBlank;
+
+/**
+ * Class EmployeeType defines an employee form.
+ */
+final class EmployeeType extends AbstractType
+{
+    use TranslatorAwareTrait;
+
+    /**
+     * @var array
+     */
+    private $languagesChoices;
+
+    /**
+     * @var array
+     */
+    private $tabChoices;
+
+    /**
+     * @var array
+     */
+    private $profilesChoices;
+
+    /**
+     * @var bool
+     */
+    private $isMultistoreFeatureActive;
+
+    /**
+     * @var string
+     */
+    private $defaultAvatarUrl;
+
+    /**
+     * @param array $languagesChoices
+     * @param array $tabChoices
+     * @param array $profilesChoices
+     * @param bool $isMultistoreFeatureActive
+     * @param string $defaultAvatarUrl
+     */
+    public function __construct(
+        array $languagesChoices,
+        array $tabChoices,
+        array $profilesChoices,
+        $isMultistoreFeatureActive,
+        $defaultAvatarUrl
+    ) {
+        $this->languagesChoices = $languagesChoices;
+        $this->tabChoices = $tabChoices;
+        $this->profilesChoices = $profilesChoices;
+        $this->isMultistoreFeatureActive = $isMultistoreFeatureActive;
+        $this->defaultAvatarUrl = $defaultAvatarUrl;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function buildForm(FormBuilderInterface $builder, array $options)
+    {
+        $builder
+            ->add('firstname', TextType::class, [
+                'constraints' => [
+                    $this->getNotBlankConstraint(),
+                    $this->getLengthConstraint(FirstName::MAX_LENGTH),
+                ],
+            ])
+            ->add('lastname', TextType::class, [
+                'constraints' => [
+                    $this->getNotBlankConstraint(),
+                    $this->getLengthConstraint(LastName::MAX_LENGTH),
+                ],
+            ])
+            ->add('email', EmailType::class, [
+                'constraints' => [
+                    $this->getNotBlankConstraint(),
+                    $this->getLengthConstraint(EmployeeEmail::MAX_LENGTH),
+                    new Email([
+                        'message' => $this->trans('This field is invalid', [], 'Admin.Notifications.Error'),
```

Comment:
Why is it considered as invalid? If it is because of the length, perhaps we should add an element to help the user know what to change, like _This field cannot be longer than %limit% characters_ in _Admin.Notifications.Error_.

Reply:
This particular message _This field is invalid_ is shown when a value entered in email field is not a valid email address (for example when @ symbol is missing).
It's not related to length - in case of too long value it currently shows "This field cannot be longer than 255 characters".

Reply:
Alright, thank you @rokaszygmantas, let's keep _This field is invalid_ then!